app/main.py

```diff
@@ -2,4 +2,71 @@ class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int | float,
+            coords: list = None
+    ) -> None:
+        if not coords:
+            coords = [0, 0]
+
+        self.name = name
+        self.weight = weight
+        self.coords = coords
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int | float,
+            coords: list = None
+    ) -> None:
+        super().__init__(name, weight, coords)
+        if not coords:
+            self.coords = [0, 0, 0]
```

Comment:
Do not redefine `coords` after passing them to the `__init__`, you could evaluate `coords` at the stage of passing them to the `__init__` using `or` expression
```suggestion
        super().__init__(name, weight, coords or [0, 0, 0])
```